packages/nextjs/src/utils/config/webpack.ts

```diff
@@ -0,0 +1,206 @@
+ * @param userSentryWebpackPluginOptions The user's SentryWebpackPlugin config, as passed to `withSentryConfig`
+ * @returns The function to set as the nextjs config's `webpack` value
+ */
+export function constructWebpackConfigFunction(
+  userNextConfig: ExportedNextConfig = {},
+  userSentryWebpackPluginOptions: Partial<SentryWebpackPluginOptions> = {},
+): WebpackConfigFunction {
+  const newWebpackFunction = (config: WebpackConfigObject, options: BuildContext): WebpackConfigObject => {
+    // if we're building server code, store the webpack output path as an env variable, so we know where to look for the
+    // webpack-processed version of `sentry.server.config.js` when we need it
+    if (config.target === 'node') {
+      storeServerConfigFileLocation(config);
+    }
+
+    let newConfig = config;
+
+    // if user has custom webpack config (which always takes the form of a function), run it so we have actual values to
+    // work with
+    if ('webpack' in userNextConfig && typeof userNextConfig.webpack === 'function') {
+      newConfig = userNextConfig.webpack(config, options);
+    }
+
+    // Ensure quality source maps in production. (Source maps aren't uploaded in dev, and besides, Next doesn't let you
+    // change this is dev even if you want to - see
+    // https://github.com/vercel/next.js/blob/master/errors/improper-devtool.md.)
+    if (!options.dev) {
+      // TODO Handle possibility that user is using `SourceMapDevToolPlugin` (see
+      // https://webpack.js.org/plugins/source-map-dev-tool-plugin/)
+      // TODO Give user option to use `hidden-source-map` ?
+      newConfig.devtool = 'source-map';
+    }
+
+    // Tell webpack to inject user config files (containing the two `Sentry.init()` calls) into the appropriate output
+    // bundles. Store a separate reference to the original `entry` value to avoid an infinite loop. (In a synchronous
+    // world, `x = () => f(x)` is fine, because the dereferencing is guaranteed to happen before the assignment, meaning
+    // we know f will get the original value of x. But in an async world, if we do `x = async () => f(x)`, the
+    // assignment happens *before* the dereferencing, meaning f is passed the new value. In other words, in that
+    // scenario, the new value is defined in terms of itself, with predictably bad consequences. Theoretically this
+    // could also be fixed by using `bind`, but this is way simpler.)
+    const origEntryProperty = newConfig.entry;
+    newConfig.entry = () => addSentryToEntryProperty(origEntryProperty, options.isServer);
+
+    // Add the Sentry plugin, which uploads source maps to Sentry when not in dev
+    checkWebpackPluginOverrides(userSentryWebpackPluginOptions);
+    newConfig.plugins = newConfig.plugins || [];
+    newConfig.plugins.push(
+      // @ts-ignore Our types for the plugin are messed up somehow - TS wants this to be `SentryWebpackPlugin.default`,
```

Comment:
Is this a `@sentry/webpack-plugin` issue? We should open a GH issue then and make sure we document that in that repo.